grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -132,9 +132,9 @@ class LocationApiController extends BaseDomainApiController {
     }
 
     def supportedActivities = {
-        def data = ActivityCode.list().collect { it.name() }
-
-        render ([data:data] as JSON)
+        boolean isAutosaveSupported = grailsApplication.config.openboxes.autosave.enabled
+        def data = isAutosaveSupported ? ActivityCode.listWithAutosave() : ActivityCode.list()
```

Comment:
I still don't like this approach, but I am not sure what would be better. I think we should tech huddle this and make an improvement/tech debt ticket for this to make it more generic and configurable for other activities. What I imagine here would be a configurable supported activities property and a filter that would add an activity to the list if it is globally enabled.

@alannadolny for now let's move this logic to the ActivityCode.list(), so it is checked only once.

cc @jmiranda